### ISSUE TITLE
Read correction does not produce output for short reads

### ISSUE THREAD

**Issue by osthomas** [marbl/canu#2182]
Hi,

I am using canu 2.2 for read correction of Nanopore amplicon reads. The majority of the amplicons is about 600 bp long, but there are instances of 300 bp amplicons.

In an initial step, I have performed clustering of individual reads. I then want to run canu correction on these individual clusters.

Based on a test data set, this seems to proceed fine, however, I find that for clusters containing 300 bp amplicons, I get no output of corrected reads (`x.correctedReads.fastq.gz`). I do get the corrected reads for the longer amplicons.

For each cluster, I use the following options, where `$cno` is the cluster number, and `$cno_dir` is a directory for the cluster sequences:

```bash
    canu -correct -p "c$cno" -d "$canu_tmpdir" \
        -nanopore-raw "$cno_dir/seqs.fasta" \
        --maxMemory=8G \
        --maxThreads=8 \
        --genomeSize=1000 \
        --minReadLength=300 \
        --minOverlapLength=300 \
        --corMinCoverage=4 \
        --useGrid=false
```

I know I am lying to canu about the `genomeSize`. Sorry.

If I set `corMinCoverage=0`, output is produced also for the 300 bp amplicons. There is definitely enough coverage though, many of the clustered sequences are even 100% identical. Interestingly, if I artificially increase the amplicon length by simply pasting each sequence behind itself (doubling the length for each record in the clustered fasta), I also get the corrected sequences.

I played around with `minReadLength`, `minOverlapLength` and `corOutCoverage` to no avail.
Am I missing an option somewhere or are sequences this short fundamentally incompatible with canu's error correction model?

Thanks in advance!

Here is the `seqStore.err` and a full log of one of the clusters which does not generate corrected output. The `No corrected reads output generated` line also shows up in a cluster with successful correction, by the way.

## seqStore.err

```
               reads               bases
---------- --------- ------ ------------ ------
Loaded         43423  94.9%     13285621  95.1%  /tmp/tmp.yNeAKW1Y0v/c0/seqs.fasta
Short           2332   5.1%       684169   4.9%


All reads processed.

               reads               bases
---------- --------- ------ ------------ ------
Loaded         43423  94.9%     13285621  95.1%
Short           2332   5.1%       684169   4.9%


EXCESSIVE COVERAGE DETECTED.  Sampling reads.

For genome size of         1000 bases,
            retain       200000 bases (200.00X coverage).

Found        43423 reads with     13285621 bases (13285.62X coverage).
Dropped      42769 reads with     13085543 bases (13085.54X coverage).
Retained       654 reads with       200078 bases (200.08X coverage).

Bye.
```

## Full log

```
-- CONFIGURE CANU
--
-- Detected Java(TM) Runtime Environment '17.0.3-internal' (from '/localenv/user/miniconda3/envs/read_correction/lib/jvm/bin/java') without -d64 support.
-- Detected gnuplot version '5.4 patchlevel 5   ' (from 'gnuplot') and image format 'png'.
--
-- Detected 128 CPUs and 252 gigabytes of memory on the local machine.
--
-- Detected Slurm with 'sinfo' binary in /package/slurm/bin/sinfo.
--          Slurm disabled by useGrid=false
--
-- Local machine mode enabled; grid support not detected or not allowed.
--
-- Job limits:
--      8 gigabytes memory  (maxMemory option).
--      8 CPUs              (maxThreads option).
--
--                                (tag)Concurrency
--                         (tag)Threads          |
--                (tag)Memory         |          |
--        (tag)             |         |          |       total usage      algorithm
--        -------  ----------  --------   --------  --------------------  -----------------------------
-- Local: meryl      4.000 GB    4 CPUs x   2 jobs     8.000 GB   8 CPUs  (k-mer counting)
-- Local: hap        4.000 GB    4 CPUs x   2 jobs     8.000 GB   8 CPUs  (read-to-haplotype assignment)
-- Local: cormhap    6.000 GB    8 CPUs x   1 job      6.000 GB   8 CPUs  (overlap detection with mhap)
-- Local: obtovl     4.000 GB    8 CPUs x   1 job      4.000 GB   8 CPUs  (overlap detection)
-- Local: utgovl     4.000 GB    8 CPUs x   1 job      4.000 GB   8 CPUs  (overlap detection)
-- Local: cor        -.--- GB    4 CPUs x   - jobs     -.--- GB   - CPUs  (read correction)
-- Local: ovb        4.000 GB    1 CPU  x   2 jobs     8.000 GB   2 CPUs  (overlap store bucketizer)
-- Local: ovs        8.000 GB    1 CPU  x   1 job      8.000 GB   1 CPU   (overlap store sorting)
-- Local: red        8.000 GB    4 CPUs x   1 job      8.000 GB   4 CPUs  (read error detection)
-- Local: oea        8.000 GB    1 CPU  x   1 job      8.000 GB   1 CPU   (overlap error adjustment)
-- Local: bat        8.000 GB    4 CPUs x   1 job      8.000 GB   4 CPUs  (contig construction with bogart)
-- Local: cns        -.--- GB    4 CPUs x   - jobs     -.--- GB   - CPUs  (consensus)
--
-- Found untrimmed raw Nanopore reads in the input files.
--
-- Generating assembly 'c0' in '/tmp/tmp.yNeAKW1Y0v/c0/canutmp':
--   genomeSize:
--     1000
--
--   Overlap Generation Limits:
--     corOvlErrorRate 0.3200 ( 32.00%)
--     obtOvlErrorRate 0.1200 ( 12.00%)
--     utgOvlErrorRate 0.1200 ( 12.00%)
--
--   Overlap Processing Limits:
--     corErrorRate    0.3000 ( 30.00%)
--     obtErrorRate    0.1200 ( 12.00%)
--     utgErrorRate    0.1200 ( 12.00%)
--     cnsErrorRate    0.2000 ( 20.00%)
--
--   Stages to run:
--     only correct raw reads.
--
--
-- BEGIN CORRECTION
----------------------------------------
-- Starting command on Fri Dec  9 18:46:06 2022 with 90.631 GB free disk space

    cd .
    ./c0.seqStore.sh \
    > ./c0.seqStore.err 2>&1

-- Finished on Fri Dec  9 18:46:07 2022 (one second) with 90.618 GB free disk space
----------------------------------------
--
-- In sequence store './c0.seqStore':
--   Found 654 reads.
--   Found 200078 bases (200.07 times coverage).
--    Histogram of raw reads:
--
--    G=200078                           sum of  ||               length     num
--    NG         length     index       lengths  ||                range    seqs
--    ----- ------------ --------- ------------  ||  ------------------- -------
--    00010          307        64        20105  ||        300                 7|-
--    00020          306       130        40311  ||        301                10|--
--    00030          306       195        60201  ||        302                 6|-
--    00040          306       260        80091  ||        303                18|---
--    00050          306       326       100287  ||        304                23|----
--    00060          306       391       120177  ||        305                55|--------
--    00070          306       456       140067  ||        306               460|---------------------------------------------------------------
--    00080          306       522       160263  ||        307                36|-----
--    00090          305       587       180100  ||        308                13|--
--    00100          300       653       200078  ||        309                 5|-
--    001.000x                 654       200078  ||        310                 4|-
--                                               ||        311                 5|-
--                                               ||        312                 6|-
--                                               ||        313                 1|-
--                                               ||        314                 0|
--                                               ||        315                 1|-
--                                               ||        316                 0|
--                                               ||        317                 0|
--                                               ||        318                 2|-
--                                               ||        319                 0|
--                                               ||        320                 1|-
--                                               ||        321                 0|
--                                               ||        322                 0|
--                                               ||        323                 1|-
--
----------------------------------------
-- Starting command on Fri Dec  9 18:46:07 2022 with 90.618 GB free disk space

    cd correction/0-mercounts
    ./meryl-configure.sh \
    > ./meryl-configure.err 2>&1

-- Finished on Fri Dec  9 18:46:07 2022 (lickety-split) with 90.618 GB free disk space
----------------------------------------
--  segments   memory batches
--  -------- -------- -------
--        01  0.01 GB       2
--
--  For 654 reads with 200078 bases, limit to 1 batch.
--  Will count kmers using 01 jobs, each using 2 GB and 4 threads.
--
-- Finished stage 'merylConfigure', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'meryl' concurrent execution on Fri Dec  9 18:46:07 2022 with 90.618 GB free disk space (1 processes; 2 concurrently)

    cd correction/0-mercounts
    ./meryl-count.sh 1 > ./meryl-count.000001.out 2>&1

-- Finished on Fri Dec  9 18:46:08 2022 (one second) with 90.615 GB free disk space
----------------------------------------
-- Found 1 Kmer counting (meryl) outputs.
-- Finished stage 'cor-merylCountCheck', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'meryl' concurrent execution on Fri Dec  9 18:46:08 2022 with 90.615 GB free disk space (1 processes; 2 concurrently)

    cd correction/0-mercounts
    ./meryl-process.sh 1 > ./meryl-process.000001.out 2>&1

-- Finished on Fri Dec  9 18:46:09 2022 (one second) with 90.616 GB free disk space
----------------------------------------
-- Meryl finished successfully.  Kmer frequency histogram:
--
--  16-mers                                                                                           Fraction
--    Occurrences   NumMers                                                                         Unique Total
--       1-     1         0                                                                        0.0000 0.0000
--       2-     2       506 ********************************************************************** 0.3822 0.0054
--       3-     4       267 ************************************                                   0.5030 0.0080
--       5-     7       104 **************                                                         0.6050 0.0110
--       8-    11        55 *******                                                                0.6692 0.0142
--      12-    16        51 *******                                                                0.7160 0.0176
--      17-    22         8 *                                                                      0.7447 0.0206
--      23-    29        29 ****                                                                   0.7553 0.0222
--      30-    37         1                                                                        0.7711 0.0251
--      38-    46         5                                                                        0.7749 0.0262
--      47-    56         1                                                                        0.7757 0.0264
--      57-    67         0                                                                        0.0000 0.0000
--      68-    79         2                                                                        0.7764 0.0268
--      80-    92         4                                                                        0.7795 0.0285
--      93-   106         0                                                                        0.0000 0.0000
--     107-   121         0                                                                        0.0000 0.0000
--     122-   137         0                                                                        0.0000 0.0000
--     138-   154         0                                                                        0.0000 0.0000
--     155-   172         0                                                                        0.0000 0.0000
--     173-   191         0                                                                        0.0000 0.0000
--     192-   211         0                                                                        0.0000 0.0000
--     212-   232         0                                                                        0.0000 0.0000
--     233-   254         0                                                                        0.0000 0.0000
--     255-   277         0                                                                        0.0000 0.0000
--     278-   301         0                                                                        0.0000 0.0000
--     302-   326         0                                                                        0.0000 0.0000
--     327-   352         0                                                                        0.0000 0.0000
--     353-   379         0                                                                        0.0000 0.0000
--     380-   407         0                                                                        0.0000 0.0000
--     408-   436         0                                                                        0.0000 0.0000
--     437-   466         0                                                                        0.0000 0.0000
--     467-   497         0                                                                        0.0000 0.0000
--     498-   529         0                                                                        0.0000 0.0000
--     530-   562        17 **                                                                     0.7817 0.0348
--     563-   596        17 **                                                                     0.7961 0.0909
--     597-   631       120 ****************                                                       0.8074 0.1376
--     632-   667       137 ******************                                                     0.9033 0.5593
--
--           0 (max occurrences)
--      186604 (total mers, non-unique)
--        1324 (distinct mers, non-unique)
--           0 (unique mers)
-- Finished stage 'meryl-process', reset canuIteration.
--
-- Removing meryl database 'correction/0-mercounts/c0.ms16'.
--
-- OVERLAPPER (mhap) (correction)
--
-- Set corMhapSensitivity=low based on read coverage of 200.07.
--
-- PARAMETERS: hashes=256, minMatches=3, threshold=0.8
--
-- Given 5.4 GB, can fit 16200 reads per block.
-- For 4 blocks, set stride to 2 blocks.
-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.
-- Configured 3 mhap precompute jobs.
-- Configured 3 mhap overlap jobs.
-- Finished stage 'cor-mhapConfigure', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'cormhap' concurrent execution on Fri Dec  9 18:46:09 2022 with 90.618 GB free disk space (3 processes; 1 concurrently)

    cd correction/1-overlapper
    ./precompute.sh 1 > ./precompute.000001.out 2>&1
    ./precompute.sh 2 > ./precompute.000002.out 2>&1
    ./precompute.sh 3 > ./precompute.000003.out 2>&1

-- Finished on Fri Dec  9 18:46:13 2022 (4 seconds) with 90.614 GB free disk space
----------------------------------------
-- All 3 mhap precompute jobs finished successfully.
-- Finished stage 'cor-mhapPrecomputeCheck', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'cormhap' concurrent execution on Fri Dec  9 18:46:13 2022 with 90.614 GB free disk space (3 processes; 1 concurrently)

    cd correction/1-overlapper
    ./mhap.sh 1 > ./mhap.000001.out 2>&1
    ./mhap.sh 2 > ./mhap.000002.out 2>&1
    ./mhap.sh 3 > ./mhap.000003.out 2>&1

-- Finished on Fri Dec  9 18:46:25 2022 (12 seconds) with 90.613 GB free disk space
----------------------------------------
-- Found 3 mhap overlap output files.
-- Finished stage 'cor-mhapCheck', reset canuIteration.
----------------------------------------
-- Starting command on Fri Dec  9 18:46:25 2022 with 90.613 GB free disk space

    cd correction
    /localenv/user/miniconda3/envs/read_correction/bin/ovStoreConfig \
     -S ../c0.seqStore \
     -M 4-8 \
     -L ./1-overlapper/ovljob.files \
     -create ./c0.ovlStore.config \
     > ./c0.ovlStore.config.txt \
    2> ./c0.ovlStore.config.err

-- Finished on Fri Dec  9 18:46:25 2022 (fast as lightning) with 90.613 GB free disk space
----------------------------------------
--
-- Creating overlap store correction/c0.ovlStore using:
--      2 buckets
--      2 slices
--        using at most 1 GB memory each
-- Finished stage 'cor-overlapStoreConfigure', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'ovB' concurrent execution on Fri Dec  9 18:46:25 2022 with 90.613 GB free disk space (2 processes; 2 concurrently)

    cd correction/c0.ovlStore.BUILDING
    ./scripts/1-bucketize.sh 1 > ./logs/1-bucketize.000001.out 2>&1
    ./scripts/1-bucketize.sh 2 > ./logs/1-bucketize.000002.out 2>&1

-- Finished on Fri Dec  9 18:46:25 2022 (fast as lightning) with 90.613 GB free disk space
----------------------------------------
-- Overlap store bucketizer finished.
-- Finished stage 'cor-overlapStoreBucketizerCheck', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'ovS' concurrent execution on Fri Dec  9 18:46:25 2022 with 90.613 GB free disk space (2 processes; 1 concurrently)

    cd correction/c0.ovlStore.BUILDING
    ./scripts/2-sort.sh 1 > ./logs/2-sort.000001.out 2>&1
    ./scripts/2-sort.sh 2 > ./logs/2-sort.000002.out 2>&1

-- Finished on Fri Dec  9 18:46:26 2022 (one second) with 90.608 GB free disk space
----------------------------------------
-- Overlap store sorter finished.
-- Finished stage 'cor-overlapStoreSorterCheck', reset canuIteration.
----------------------------------------
-- Starting command on Fri Dec  9 18:46:26 2022 with 90.608 GB free disk space

    cd correction
    /localenv/user/miniconda3/envs/read_correction/bin/ovStoreIndexer \
      -O  ./c0.ovlStore.BUILDING \
      -S ../c0.seqStore \
      -C  ./c0.ovlStore.config \
      -delete \
    > ./c0.ovlStore.BUILDING.index.err 2>&1

-- Finished on Fri Dec  9 18:46:26 2022 (furiously fast) with 90.609 GB free disk space
----------------------------------------
-- Overlap store indexer finished.
-- Checking store.
----------------------------------------
-- Starting command on Fri Dec  9 18:46:26 2022 with 90.609 GB free disk space

    cd correction
    /localenv/user/miniconda3/envs/read_correction/bin/ovStoreDump \
     -S ../c0.seqStore \
     -O  ./c0.ovlStore \
     -counts \
     > ./c0.ovlStore/counts.dat 2> ./c0.ovlStore/counts.err

-- Finished on Fri Dec  9 18:46:26 2022 (in the blink of an eye) with 90.609 GB free disk space
----------------------------------------
--
-- Overlap store 'correction/c0.ovlStore' successfully constructed.
-- Found 6212 overlaps for 641 reads; 42782 reads have no overlaps.
--
--
-- Purged 0.004 GB in 9 overlap output files.
-- Finished stage 'cor-createOverlapStore', reset canuIteration.
-- Computing correction layouts.
--   Local  filter coverage   80
--   Global filter coverage   40
----------------------------------------
-- Starting command on Fri Dec  9 18:46:26 2022 with 90.614 GB free disk space

    cd correction
    /localenv/user/miniconda3/envs/read_correction/bin/generateCorrectionLayouts \
      -S ../c0.seqStore \
      -O  ./c0.ovlStore \
      -C  ./c0.corStore.WORKING \
      -eC 80 \
      -xC 40 \
    > ./c0.corStore.err 2>&1

-- Finished on Fri Dec  9 18:46:26 2022 (like a bat out of hell) with 90.612 GB free disk space
----------------------------------------
-- Finished stage 'cor-buildCorrectionLayoutsConfigure', reset canuIteration.
-- Computing correction layouts.
----------------------------------------
-- Starting command on Fri Dec  9 18:46:26 2022 with 90.612 GB free disk space

    cd correction/2-correction
    /localenv/user/miniconda3/envs/read_correction/bin/filterCorrectionLayouts \
      -S  ../../c0.seqStore \
      -C     ../c0.corStore \
      -R      ./c0.readsToCorrect.WORKING \
      -cc 4 \
      -cl 300 \
      -g  1000 \
      -c  40 \
    > ./c0.readsToCorrect.err 2>&1

-- Finished on Fri Dec  9 18:46:26 2022 (lickety-split) with 90.609 GB free disk space
----------------------------------------
--                             original      original
--                            raw reads     raw reads
--   category                w/overlaps  w/o/overlaps
--   -------------------- ------------- -------------
--   Number of Reads                387         43036
--   Number of Bases             118464         77630
--   Coverage                   118.464        77.630
--   Median                         306             0
--   Mean                           306             1
--   N50                            306           306
--   Minimum                        300             0
--   Maximum                        323           312
--
--                                        --------corrected---------  ----------rescued----------
--                             evidence                     expected                     expected
--   category                     reads            raw     corrected            raw     corrected
--   -------------------- -------------  ------------- -------------  ------------- -------------
--   Number of Reads                 96            137           137              0             0
--   Number of Bases              29440          41969         40188              0             0
--   Coverage                    29.440         41.969        40.188          0.000         0.000
--   Median                         306            306           293              0             0
--   Mean                           306            306           293              0             0
--   N50                            306            306           293              0             0
--   Minimum                        306            306           293              0             0
--   Maximum                        323            320           307              0             0
--
--                        --------uncorrected--------
--                                           expected
--   category                       raw     corrected
--   -------------------- ------------- -------------
--   Number of Reads              43286         43286
--   Number of Bases             154125         55101
--   Coverage                   154.125        55.101
--   Median                           0             0
--   Mean                             3             1
--   N50                            306           293
--   Minimum                          0             0
--   Maximum                        323           310
--
--   Maximum Memory           527639228
-- Finished stage 'cor-filterCorrectionLayouts', reset canuIteration.
--
-- Correction jobs estimated to need at most 0.491 GB for computation.
-- Correction jobs will request 6 GB each.
--
-- Local: cor        6.000 GB    4 CPUs x   1 job      6.000 GB   4 CPUs  (read correction)
--
--
-- Configuring correction jobs:
--   Reads estimated to need at most 0.491 GB for computation.
--   Jobs will request 6 GB each.
----------------------------------------
-- Starting command on Fri Dec  9 18:46:26 2022 with 90.609 GB free disk space

    cd correction/2-correction
    ./correctReadsPartition.sh \
    > ./correctReadsPartition.err 2>&1

-- Finished on Fri Dec  9 18:46:27 2022 (one second) with 90.609 GB free disk space
----------------------------------------
-- Finished stage 'cor-generateCorrectedReadsConfigure', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'cor' concurrent execution on Fri Dec  9 18:46:27 2022 with 90.609 GB free disk space (1 processes; 1 concurrently)

    cd correction/2-correction
    ./correctReads.sh 1 > ./correctReads.000001.out 2>&1

-- Finished on Fri Dec  9 18:46:27 2022 (furiously fast) with 90.609 GB free disk space
----------------------------------------
-- Found 1 read correction output files.
-- Finished stage 'cor-generateCorrectedReadsCheck', reset canuIteration.
-- Found 1 read correction output files.
-- Finished stage 'cor-generateCorrectedReadsCheck', reset canuIteration.
--
-- Loading corrected reads into corStore and seqStore.
----------------------------------------
-- Starting command on Fri Dec  9 18:46:27 2022 with 90.609 GB free disk space

    cd correction
    /localenv/user/miniconda3/envs/read_correction/bin/loadCorrectedReads \
      -S ../c0.seqStore \
      -C ./c0.corStore \
      -L ./2-correction/corjob.files \
    >  ./c0.loadCorrectedReads.log \
    2> ./c0.loadCorrectedReads.err

-- Finished on Fri Dec  9 18:46:27 2022 (fast as lightning) with 90.606 GB free disk space
----------------------------------------
--
-- No corrected reads generated; correctReads output saved.
--
-- Purging overlaps used for correction.
-- Finished stage 'cor-loadCorrectedReads', reset canuIteration.
----------------------------------------
-- Starting command on Fri Dec  9 18:46:27 2022 with 90.61 GB free disk space

    cd .
    /localenv/user/miniconda3/envs/read_correction/bin/sqStoreDumpFASTQ \
      -corrected \
      -S ./c0.seqStore \
      -o ./c0.correctedReads.gz \
      -fasta \
      -nolibname \
    > c0.correctedReads.fasta.err 2>&1

-- Finished on Fri Dec  9 18:46:27 2022 (furiously fast) with 90.61 GB free disk space
----------------------------------------
--
-- Corrected reads saved in 'c0.correctedReads.fasta.gz'.
-- Finished stage 'cor-dumpCorrectedReads', reset canuIteration.
--
-- Trimming skipped; not enabled.
--
-- Unitigging skipped; not enabled.
--
-- Bye.
```

**Comment by gringer** [comment]
Has your playing around with overlap and minimum read length included setting the minimum overlap length to less than the minimum read length (e.g. 150bp)? Correction can only happen if multiple reads overlap and share sequence, and if the minimum overlap length is the same size or greater than the read length, then it's impossible for any sequence to be shared.

**Comment by osthomas** [comment]
I have just tried that (`--minReadLength=300 --minOverlapLength=150`), with no change in results. For this particular cluster, the majority of reads have a length of 306, and most of these are identical, so there should be plenty of overlap.

**Comment by osthomas** [comment]
Thank you very much for the suggestions. Common kmers being excluded sounded reasonable, but unfortunately, none of the options helped.

I checked `0-mercounts/*.ms??.ignore.gz`. With the original options, the file was already more or less empty, containing just `0\t0`. Setting `corMhapFilterThreshold=0` did not change anything.

`--corOverlapper=ovl --ovlMerSize=16 --ovlMerThreshold=9999` ran, but again did not produce corrected output for the cluster in question. I also tried increasing `--ovlMerThreshold` to no effect.

`--corOverlapper=minimap` also ran, but no output.

**Comment by gringer** [comment]
The [FAQ](https://canu.readthedocs.io/en/latest/faq.html#can-i-assemble-amplicon-sequence-data) suggests that contig filtering should be turned off for amplicons using the following options:

    contigFilter="2 0 1.0 0.5 0"

> The [contigFilter](https://canu.readthedocs.io/en/latest/parameter-reference.html#contigfilter) parameter sets several parameters that control how small or low coverage initial contigs are handled. By default, initial contigs with more than 50% of the length at less than 3X coverage will be classified as ‘unassembled’ and removed from the assembly, that is, contigFilter="2 0 1.0 0.5 3". The filtering can be disabled by changing the last number from ‘3’ to ‘0’ (meaning, filter if 50% of the contig is less than 0X coverage).

**Comment by skoren** [comment]
I don't think any overlap or contig settings will help since it looks like you have overlaps from the report. It seems the reads are failing to align using the final correction step (or not ending up with enough coverage). You said corMinCoverage=0 works, does corMinCoverage=2 or 3 work as well? What about corErrorRate, setting it to 0.5 to increase tolerance for errors in the short overlaps. You don't have to re-run the whole pipeline to test these, you can just edit the correctReads.sh shell script and re-run it, removing the files in results between runs.

**Comment by osthomas** [comment]
> `contigFilter="2 0 1.0 0.5 0"`

This did not help, either.

> I do remember issues with overlap fraction being too high when I was playing around with amplicon assembly in the past.

Would this issue not also manifest after I artificially concatenated the reads to double their length?

> You said corMinCoverage=0 works, does corMinCoverage=2 or 3 work as well? 

No - it works with `0`, but even with `1`, there are no longer any corrected reads.

> What about corErrorRate, setting it to 0.5 to increase tolerance for errors in the short overlaps. 

No dice here, either. I went as high as 0.9 without effect.


Thanks for all the input, much appreciated!

**Comment by skoren** [comment]
I think the cause is this line of code:
https://github.com/marbl/canu/blame/master/src/correction/falconConsensus-alignTag.C#L443

It will skip all overlaps < 500bp with this setting. I think the 500 should be minOlapLength instead. You could try changing your code (if you compiled locally) and re-running. That would also explain why doubling your reads makes it output results.

**Comment by osthomas** [comment]
That helped!
I have built canu v2.2, master compiled but threw a segmentation fault (gcc 11.3.0).

I changed the `500` to `minOlapLength`. This worked, eventually.
However, initially, I only got very few corrected reads (3-5), despite the large input size and a requested output coverage of 40. I had to lower the `minReadLength` a little because the corrected reads were shorter (293) than my initial threshold of 300. I'm not really sure why, to be honest, because the absolute majority of the reads in this cluster are identical and have a length of 306 - but that is another issue.

Thank you all very much for your suggestions!

Is there a chance that this change will find its way into a release soon, or should I rather continue to build from source for now?
